### ISSUE TITLE
Generate complex Json schemas with BeanOutputConverter

### ISSUE THREAD

**Issue by fjtorres** [spring-projects/spring-ai#794]
**Expected Behavior**

Be able to generate more complex schemas with the `BeanOutputConverter` class, for instance with constrains definitions like max elements in array and max or min length for strings. This can be handled adding support to the definition of what modules should be loaded for _jsonschema-generator_ library. Could be allowing override the private method `generateSchema` or adding it as a constructor parameter.

**Current Behavior**

Right now the `BeanOutputConverter` is adding just the _jackson-module_ to the configuration of _jsonschema-generator_ library, this is a great starting point to have a schema definition but does not allow the definition of more complex scenarios. The code to generate the value for jsonSchema field is private so we cannot override it to adding more modules to the libray.

**Context**

Under some circumstances we can ask to the AI model to generate content with specific rules, for instance if we want to generate specific number of answers for one question. In that case we can specify it with the prompt but probably could be a good idea add it as a constraint to the expected schema just to be sure the AI model respect it. This can be handled with the maven library _jsonschema-module-jakarta-validation_. I tested locally and just add it to the  _jsonschema-generator_ library configuration it's enough to add constraints to the generated schema.

**Comment by Seol-JY** [comment]
@csterwa 

Is this feature still not implemented? I'd like to contribute to add Jakarta Validation support to `BeanOutputConverter` for constraint annotations like `@Size`, `@Min`, etc.

Would a PR be welcome? I'm thinking of adding support for `jsonschema-module-jakarta-validation` while maintaining backward compatibility.
I know this might not be easy for a beginner like me, but I'd like to give it a try!

Thanks!

**Comment by csterwa** [comment]
It is great to see your interest in providing a PR for this. There is a `help wanted` label on this issue. I'm going summon @tzolov or @markpollack to provide guidance on the approach, if any?

**Comment by Seol-JY** [comment]
Hi @csterwa @tzolov @markpollack,

Before implementing Jakarta Validation support for `BeanOutputConverter`, I'd like to get guidance on the dependency approach.

**Question**: Would it be appropriate to include `jsonschema-module-jakarta-validation` as a transitive dependency in `spring-ai-model`?

**Considerations**:
- This would enable validation constraints in JSON schemas automatically
- However, it adds a dependency that only benefits users who actually use Bean Validation annotations
- Spring Boot moved validation to a separate starter (`spring-boot-starter-validation`) for similar reasons

**Alternative**: Auto-detection pattern - only enable validation schema features when the module is already on the user's classpath.

What would be the preferred approach from the Spring AI team's perspective?

Thanks for your guidance!

**Comment by s-jepsen** [comment]
Any progress on this issue?

I think it's a real relevant usecase.

**Comment by Seol-JY** [comment]
@s-jepsen I agree this is a relevant use case. Currently exploring potential implementation approaches.
